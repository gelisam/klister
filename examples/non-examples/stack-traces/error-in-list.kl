#lang "prelude.kl"

(import "defun.kl")
(import "list.kl")

(define thing 'nothing)
(define the-error (error 'Im-an-error))

(defun fail (thing) (+ 1 thing))

-- (example `(list-syntax (,thing (fail the-error) ()) thing))
(example `('a 'b ,the-error))