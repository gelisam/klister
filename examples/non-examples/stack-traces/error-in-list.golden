In Phase: p0:
    error-in-list.kl:7.27-7.38: Im-an-error
    stack trace:
    ---- Halt
